noxfile.py

```diff
@@ -59,37 +53,30 @@ def get_requirements_filepath(
 
 @nox.session
 def requirements(session):
-    """Regenerate pinned requirements files used during CI."""
+    """Regenerate pinned requirements files."""
 
     session.install("uv >= 0.1.39")
 
-    category_version_resolution = [
-        ("all", maxpython, "highest"),
-        ("docs", maxpython, "highest"),
-        ("tests", minpython, "lowest-direct"),
-    ]
-
-    category_version_resolution += [
-        ("tests", version, "highest") for version in supported_python_versions
+    category_version_resolution: list[tuple[str, str, str]] = [
+        ("tests", version, resolution)
+        for version in supported_python_versions
+        for resolution in ("highest", "lowest-direct")
     ]
 
-    category_flags = {
+    category_flags: dict[str, tuple[str, ...]] = {
```

Comment:
📝 An annotation of `tuple[str]` would indicate that the tuple must contain exactly one string, whereas `tuple[str, ...]` indicates any number of strings.  The `...` is not necessary for lists, since lists are mutable.